### ISSUE TITLE
PA2 level5 peek method

### ISSUE THREAD

**Issue by aarontxz** [nus-cs2030/2223-s2#915]
After changing T problem to Supplier<T> problem wont the peek function need to be changed? The walkthrough does not seem to address this issue am i missing something?

**Comment by Huanqing-Mao** [comment]
Your peek method will have to take out the value from the supplier

**Comment by aarontxz** [comment]
if my peek calls a value out of the supplier wouldtn that affect my solve later on when i try to get another value our of the supplier?

**Comment by Huanqing-Mao** [comment]
Peek should not affect your solve methods, as the solve() and solve(bop) shouldn't rely on your peek.

peek() is just for you to see what is inside the DnC object, and it will not be needed for computation

**Comment by EvanCh08** [comment]
Your peek should not affect your solve method, it is used just to see what is inside your supplier, so do .get() in your peek method and you will be okay.

**Comment by NwaterM** [comment]
Nope, your peek function should still take in T problem, but instead you need to call get(). in the method. This is because the problem is now coming from a Supplier instead.

**Comment by stxe09** [comment]
It's okay to use "extra" get() here since it's almost like a terminating call, so it won't affect any other functions

**Comment by ryantangmj** [comment]
You just have to use .get() to get rid of the Supplier wrapping around T

**Comment by mkim123** [comment]
just add a .get() method to your this.problem. The walkthrough does not seem to cover everything

**Comment by jarenl0ke** [comment]
The solve() function does not rely on peek() so it should be ok to use .get() for the problem in peek(). It seems that peek() is only used for the first Level.

**Comment by koojaylynn** [comment]
hello! you should just add .get() to the this.problem to get the T value from the Supplier :)

**Comment by slimzaza** [comment]
peek() is independent of the other methods like solve, so u can go ahead and use .get() in the solve() method

**Comment by tjdbryan** [comment]
.get() to extract the value from Supplier

**Comment by NeoHW** [comment]
you could try using .get()!

**Comment by sohjoana** [comment]
instead of this.problem replace it with this.problem.get()

**Comment by Cfyjdfk** [comment]
Yup, walkthrough isn't exactly complete. Use .get()

**Comment by sushimakerr** [comment]
your peek method should call this.problem.get()

**Comment by e0969533** [comment]
in peek method, u can use get() to take out T from the supplier !

**Comment by Aron113** [comment]
you could use .get() to get the value

**Comment by shortlion99** [comment]
yup just use .get() to access T from the supplier :-)

**Comment by redcolorbicycle** [comment]
walkthrough has many parts missing, use .get() to access T

**Comment by vVicksv** [comment]
use .get() since it's a supplier class to get the value in it.

**Comment by territxl** [comment]
the peek is meant to output the evaluation and hence, since the problem is of a supplier type, calling get and passing it into the consumer accept() would work :)

**Comment by haqanh** [comment]
now that you have changed your problem to a supplier, u should call .get such that peek will still return the problem itself!

**Comment by koiljat** [comment]
call get() to get the value, since problem is now a supplier

**Comment by 12thfret** [comment]
You need to call .get() to get the value!
